geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/FilterHelper.scala

```diff
@@ -337,15 +337,18 @@ object FilterHelper {
         try {
           val prop = f.getExpression.asInstanceOf[PropertyName].getPropertyName
           if (prop != attribute) { FilterValues.empty } else {
-            // Remove the trailing wildcard and create a range prefix
+            // find the first wildcard and create a range prefix
             val literal = f.getLiteral
-            val lower = if (literal.endsWith(MULTICHAR_WILDCARD)) {
-              literal.substring(0, literal.length - MULTICHAR_WILDCARD.length)
-            } else {
-              literal
+            var i = literal.indexWhere(Wildcards.contains)
+            // check for escaped wildcards
+            while (i > 1 && literal.charAt(i - 1) == '\\' && literal.charAt(i - 2) == '\\') {
```

Comment:
@jnh5y does this seem like the right escape behavior?

Reply:
seems to be... https://github.com/geotools/geotools/blob/master/modules/library/main/src/main/java/org/geotools/filter/LikeFilterImpl.java#L47